spikeinterface/sorters/runsorter.py

```diff
@@ -164,116 +156,112 @@ def run_sorter_docker(sorter_name, recording, docker_image, output_folder=None,
-            # install from github source several stuff
+            # TODO later check output
             cmd = 'pip install --upgrade --force MEArec'
-            commands.append(cmd)
-
+            res = container.exec_run(cmd)
             cmd = 'pip install -e git+https://github.com/SpikeInterface/spikeinterface.git#egg=spikeinterface[full]'
-            commands.append(cmd)
-
+            res = container.exec_run(cmd)
             cmd = 'pip install --upgrade --force https://github.com/NeuralEnsemble/python-neo/archive/master.zip'
-            commands.append(cmd)
+            res = container.exec_run(cmd)
         else:
-            # install from pypi with same version as host
             if verbose:
                 print(f"Installing spikeinterface=={si_version} in {docker_image}")
             cmd = f'pip install --upgrade --force spikeinterface[full]=={si_version}'
-            commands.append(cmd)
+            res = container.exec_run(cmd)
+    else:
+        # TODO version checking
+        if verbose:
+            print(f'spikeinterface is already installed in {docker_image}')
+
 
     # run sorter on folder
-    cmd = 'python /sorting_output_folder/in_docker_sorter_script.py'
-    commands.append(cmd)
+    if verbose:
+        print(f'Running {sorter_name} sorter inside {docker_image}')
+    cmd = 'python "{}"'.format(parent_folder/'in_docker_sorter_script.py')
+    res = container.exec_run(cmd)
+    run_sorter_output = res.output
 
-    # put file permission to user (because docker is root...)
+    # chown folder to user uid
     uid = os.getuid()
-    cmd = f'chown {uid}:{uid} -R /sorting_output_folder/{folder_name}'
-    commands.append(cmd)
-
-    # ~ commands = commands[0:4]
-    # ~ commands = ' ; '.join(commands)
-    commands = ' && '.join(commands)
-    command = f'sh -c "{commands}"'
-    #  print(command)
-
-    extra_kwargs = {}
-    if SorterClass.docker_requires_gpu:
-        extra_kwargs["device_requests"] = [docker.types.DeviceRequest(count=-1, capabilities=[['gpu']])]
-
+    cmd = f'chown {uid}:{uid} -R "{output_folder}"'
+    res = container.exec_run(cmd)
+    
     if verbose:
-        print(f"Running sorter in {docker_image}")
-        # flush whatever is in the stdout
-        sys.stdout.flush()
-
-    try:
-        res = client.containers.run(docker_image, command=command, volumes=volumes, 
-                                    **extra_kwargs)
-        # clean useless files
-        os.remove(parent_folder / 'in_docker_recording.json')
-        os.remove(parent_folder / 'in_docker_params.json')
-        os.remove(parent_folder / 'in_docker_sorter_script.py')
-    except Exception as e:
-        # clean useless files
-        os.remove(parent_folder / 'in_docker_recording.json')
-        os.remove(parent_folder / 'in_docker_params.json')
-        os.remove(parent_folder / 'in_docker_sorter_script.py')
+        print('stopping container')
```

Comment:
```suggestion
        print('Stopping container')
```

@samuelgarcia to be consistent I would also add a `print('Starting container')` when the container is created